installer/win/WixSharpInstaller/Symphony.cs

```diff
@@ -109,16 +111,30 @@ static public void Main(string[] args)
             // The property values to compare against can be found here:
             //    https://docs.microsoft.com/en-us/windows/win32/msi/operating-system-property-values
             new LaunchCondition("VersionNT>=600 AND WindowsBuild>=6001", "OS not supported"),
-            
+
             // Add registry entry used by protocol handler to launch symphony when opening symphony:// URIs
-            new RegValue(WixSharp.RegistryHive.ClassesRoot, productName + @"\shell\open\command", "", "\"[INSTALLDIR]Symphony.exe\" \"%1\"")            
+            new RegValue(WixSharp.RegistryHive.ClassesRoot, productName + @"\shell\open\command", "", "\"[INSTALLDIR]Symphony.exe\" \"%1\"")
         );
 
         // The build script which calls the wix# builder, will be run from a command environment which has %SYMVER% set.
         // So we just extract that version string, create a Version object from it, and pass it to out project definition.
         var version = System.Environment.GetEnvironmentVariable("SYMVER");
         project.Version = new System.Version(version);
 
+        // To get the correct behaviour with upgrading the product, the product GUID needs to be different for every build,
+        // but the UpgradeCode needs to stay the same. If we wanted to make a new major version and allow it to be installed
+        // side-by-side with the previous version, we would generate a new UpgradeCode for the new version onwards.
+        // More details can be found in this stackoverflow post:
+        //      https://stackoverflow.com/a/26344742
+        project.GUID = System.Guid.NewGuid();
+        project.UpgradeCode = new System.Guid("{a9b448c9-f065-41a4-87c3-da527c6a389b}");
+
+        // Don't allow installation of earlier versions, but do allow installing the same version as is already installed.
+        project.MajorUpgrade = new MajorUpgrade();
+        project.MajorUpgrade.AllowDowngrades = false;
+        project.MajorUpgrade.AllowSameVersionUpgrades = true;
+        project.MajorUpgrade.DowngradeErrorMessage = "A more recent version of Symphony is already installed on this computer.";
```

Comment:
So, in this case, if a user wants to downgrade, they'll need to uninstall the app first and then install the older version. Is that correct?

Reply:
yes, I believe that is how the current one works also